### PR TITLE
add option to submit a template through `list_search_base_path`

### PR DESCRIPTION
**Pull request by ekneg54**
* adds the feature to list_comparison processor to submit a template like `"http://${LOGPREP_TEST_TARGET}/${LOGPREP_LIST_SEARCH_BASE_PATH}/${LOGPREP_LIST}?ref=bla"` in `list_search_base_path` config option

### CODE REVIEW

File: logprep/processor/list_comparison/rule.py [hunk 1]
```diff
@@ -78,6 +79,16 @@ def _get_list_search_base_path(self, list_search_base_path):
     def init_list_comparison(self, list_search_base_path: Optional[str] = None):
         """init method for list_comparision lists"""
         list_search_base_path = self._get_list_search_base_path(list_search_base_path)
+        if list_search_base_path.startswith("http"):
```
Comment (urdit52) [inline logprep/processor/list_comparison/rule.py:82]: As this behaviour seems to always apply when a base path starting with http is used (and in order to make this accessible), could you please also add the template functionality to the processor documentation?

**Review by urdit52 — CHANGES_REQUESTED** [review]
Thanks for submitting. Could you please also add the template functionality to the documentation?

File: logprep/processor/list_comparison/rule.py (hunk 1 above)
Comment (ekneg54) [reply logprep/processor/list_comparison/rule.py:84]: DONE

File: logprep/processor/list_comparison/processor.py [hunk 2]
```diff
@@ -45,7 +45,10 @@ class Config(Processor.Config):
 
         list_search_base_path: str = field(validator=validators.instance_of(str))
         """Relative list paths in rules will be relative to this path if this is set.
-        This parameter is optional. For string format see :ref:`getters`."""
+        This parameter is optional. For string format see :ref:`getters`.
+        You can pass a template with keys from environment,
+        like :code:`${<your environemnt variable>}`. The special key :code:`${LOGPREP_LIST}`
+        will be filled by this processor."""
```
Comment (urdit52) [inline logprep/processor/list_comparison/processor.py:51]:
```suggestion
        You can also pass a template with keys from environment,
        e.g.,  :code:`${<your environment variable>}`. The special key :code:`${LOGPREP_LIST}`
        will be filled by this processor. """
```

File: logprep/processor/list_comparison/rule.py [hunk 3]
```diff
@@ -63,7 +63,9 @@ class Config(FieldManagerRule.Config):
         """List of files. For string format see :ref:`getters`."""
         list_search_base_path: str = field(validator=validators.instance_of(str), factory=str)
         """Base Path from where to find relative files from :code:`list_file_paths`.
-        For string format see :ref:`getters`. (Optional)"""
+        For string format see :ref:`getters`. You can pass a template with keys from environment,
+        like :code:`${<your environemnt variable>}`. The special key :code:`${LOGPREP_LIST}`
+        will be filled by this processor. (Optional)"""
```
Comment (urdit52) [inline logprep/processor/list_comparison/rule.py:68]:
```suggestion
        You can also pass a template with keys from environment,
        e.g.,  :code:`${<your environment variable>}`. The special key :code:`${LOGPREP_LIST}`
        will be filled by this processor. """
```
```

**Review by urdit52 — APPROVED** [review]
Please accept my minor suggestions before merging.

**Pull request merged**